chat-server/index.html

```diff
@@ -1,24 +1,123 @@
-        <label
-          >Message: <input type="text" name="text" placeholder="The message..."
-        /></label>
+      <form id="message-form" action="/messages" method="post">
+        <label>Name: <input id="new-message-from" type="text" name="from" placeholder="Your Name" /></label>
+        <label>Message: <input id="new-message-text" type="text" name="text" placeholder="The message..." /></label>
         <button type="submit">Send</button>
       </form>
-    </main>
-    <footer>
-      <h3><a href="/messages">See all messages</a></h3>
-    </footer>
-  </body>
-</html>
+    </section>
+    <section id="messages">
+      <h3><a href="" onclick="fetchAndLoadMessages()">See all messages</a></h3>
+      <div>
+        <h3>See message by ID</h3>
+        <input id="find-message-id" type="text" placeholder="Type an ID here">
+        <button onclick="fetchAndLoadMessageById()">Show message</button>
+      </div>
+      <ul id="messages-list">
+
+      </ul>
+    </section>
+  </main>
+  <footer>
+    <!-- <p>Project by Pedro Ricciardi, as part of the CYF Course - Servers Module - Chat-Server Issue</p> -->
+  </footer>
+
+  <!-- SCRIPTS -->
+  <script>
+    const renderMessage = (message) => {
+      const listItem = document.createElement('li');
+      const listItemParagraph = document.createElement('p');
+      const listItemDeleteButton = document.createElement('button');
+
+      listItemDeleteButton.textContent = '🗙';
+      listItemDeleteButton.setAttribute('class', 'delete-button');
+      listItemDeleteButton.setAttribute('onclick', 'deleteMessage(event)');
+
+      listItemParagraph.textContent = `${message.from}: ${message.text}`;
+
+      listItem.appendChild(listItemParagraph);
+      listItem.appendChild(listItemDeleteButton);
+      listItem.setAttribute('data-messageId', message.id);
+
+      return listItem;
+    };
+
+    const loadMessages = (messagesToRender) => {
+      const messagesList = document.getElementById('messages-list');
+      messagesList.innerHTML = '';
+      messagesToRender.forEach(message => {
+        const listItem = renderMessage(message);
+        messagesList.appendChild(listItem);
+      });
+    };
+
+    const loadMessagesFailure = (error) => {
+      const messagesList = document.getElementById('messages-list');
+      messagesList.innerHTML = '';
+      const errorMessage = document.createElement('p');
+      errorMessage.setAttribute('class', 'error-message');
+      errorMessage.textContent = `${error}`;
+      messagesList.appendChild(errorMessage);
+    };
+
+    const fetchAndLoadMessageById = () => {
+      const messageId = document.getElementById('find-message-id').value;
+      fetch(`/messages/${messageId}`)
+        .then(response => response.json())
+        .then(message => {
+          if (message && message.from && message.text) {
+            loadMessages([message]);
+          } else {
+            loadMessagesFailure('Message not found. Please select other ID');
```

Comment:
Good job making sure the user sees when something is wrong in the page!

Reply:
Thank you.